### PR TITLE
Add stash 3.8 automated setup options

### LINKED ISSUE #63: Add paramaters for stash 3.8 Automated setup options
Previously, even when installing Stash automatically, you still had to visit the Stash Setup Wizard in the browser to set the admin account, the Base URL and the external database connection, and to add the Stash license. 

Now, by specifying all those settings in the Stash config properties file, you can completely install and set up a Stash instance automatically, without manual intervention. Stash will read the file, and apply those settings automatically after launching.

Its documented here:
https://confluence.atlassian.com/display/STASH/Automated+setup+for+Stash

### PR DESCRIPTION
**Pull request by nanliu**
Fix #63

### CODE REVIEW

**Comment by nanliu** [comment]
Sorry, I did not have opportunity to test if this actually automates the install.

File: templates/stash-config.properties.erb [hunk 1]
```diff
@@ -1,3 +1,12 @@
+<% if scope.function_versioncmp([scope['stash::version'], '3.8.0']) >= 0 -%>
+setup.displayName=<%= scope['stash::display_name'] %>
+setup.baseUrl=<%= scope['stash::base_url'] %>
+setup.license=<%= scope['stash::license'] %>
+setup.sysadmin.username=<%= scope['stash::sysadmin_username'] %>
+setup.sysadmin.password=<%= scope['stash::sysadmin_password'] %>
+setup.sysadmin.displayName=<%= scope['stash::sysadmin_name'] %>
+setup.sysadmin.emailAddress=<%= scope['stash::sysadmin_email'] %>
```
Comment (igalic) [inline templates/stash-config.properties.erb:8]: @nanliu would you mind briefly explaining what the difference is between `scope['x::y']`, and `scope.lookupvar('x::y')`?

Comment (nanliu) [inline templates/stash-config.properties.erb:8]: It's the same thing, and available in puppet 3+, I can switch it to scope.lookupvar.

Comment (igalic) [inline templates/stash-config.properties.erb:8]: nope, it's fine, thank you!

**Pull request merged**